### PR TITLE
fixes api auth bug in techdocs backend

### PR DESCRIPTION
**Pull request by erikxiv**
Signed-off-by: Erik Larsson <erik.larsson@schibsted.com>

## Hey, I just made a Pull Request!

DefaultTechDocsCollator does not supply an api token when fetching search indices for tech docs, resulting in 401 responses for backends requiring api tokens and thus failed search indexing. This PR adds the api token to the requests to fix that.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — APPROVED** [review]

File: .changeset/seven-tomatoes-smash.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-techdocs-backend': minor
```
Comment (iamEAP) [inline .changeset/seven-tomatoes-smash.md:2]: Probably just a `patch` here, eh?

File: .changeset/seven-tomatoes-smash.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+fixes api auth bug in techdocs backend
```
Comment (iamEAP) [inline .changeset/seven-tomatoes-smash.md:5]: Probably good to mention this bug only affected the `DefaultTechDocsCollator` and not the backend itself.

**Review by iamEAP — COMMENTED** [review]
Thanks @erikxiv!  Super minor items on the changeset, but otherwise looks good to me.

**Review by iamEAP — APPROVED** [review]
🙏  Thanks again!

**Pull request merged**